src/main/java/org/spongepowered/common/util/gen/ArrayMutableBlockBuffer.java

```diff
@@ -39,25 +40,28 @@
 import org.spongepowered.common.world.extent.MutableBlockViewTransform;
 import org.spongepowered.common.world.extent.UnmodifiableBlockVolumeWrapper;
 import org.spongepowered.common.world.extent.worker.SpongeMutableBlockVolumeWorker;
+import org.spongepowered.common.world.schematic.BimapPalette;
+import org.spongepowered.common.world.schematic.GlobalPalette;
 
 public class ArrayMutableBlockBuffer extends AbstractBlockBuffer implements MutableBlockVolume {
 
+    private static final int LOCAL_PALETTE_THRESHOLD = 256;
+
     @SuppressWarnings("ConstantConditions")
     private static final BlockState AIR = BlockTypes.AIR.getDefaultState();
 
     private final BlockPalette palette;
     private final BackingData data;
 
-    public ArrayMutableBlockBuffer(BlockPalette palette, Vector3i start, Vector3i size, BackingDataType type) {
-        super(start, size);
-        this.palette = palette;
-        this.data = type.create(size);
+    public ArrayMutableBlockBuffer(Vector3i start, Vector3i size) {
+        this(size.getX() * size.getY() * size.getZ() > LOCAL_PALETTE_THRESHOLD ?
```

Comment:
Right now, remax() does not need to deal with the palettes and stuff... It could use a callback and see if it should translate the ids also... But there is also a set() in flight that will also need to be translated at that point.
The way Mojang handles this is by putting a callback in the palette. (idk if I like that)

Another way to think of the problem... The local palette has gotten so full that it is almost as big as the global palette making it no longer worth having a local palette. This problem is not specific to the block buffer, but can happen to any local palette. :/
There has to be a better way to design all this, so its clean to implement.